auth/service.bal

```diff
@@ -0,0 +1,101 @@
+import ballerina/http;
+import ballerina/crypto;
+import ballerina/uuid;
+import e_backend.data;
+import ballerina/persist;
+import ballerina/sql;
+
+listener http:Listener voterListener = new (9090);
+
+final data:Client dbClient = check new ();
+
+
+public type VoterLogin record { 
+    string nationalId; 
+    string password; 
+};
+
+service /Voter\-registration/api/v1 on voterListener {
+
+    // Register a new voter
+    @http:ResourceConfig {}
+    resource function post voters/register(@http:Payload data:Voter newVoter)
+            returns http:Created|http:Forbidden|error {
+
+        string hashedPassword = crypto:hashMd5(newVoter.password.toBytes()).toBase16();
+
+        data:VoterInsert voterInsert = {
+            id: generateId(),
+            nationalId: newVoter.nationalId,
+            fullName: newVoter.fullName,
+            mobileNumber: newVoter.mobileNumber,
+            dob: newVoter.dob,
+            gender: newVoter.gender,
+            nicChiefOccupant: newVoter.nicChiefOccupant,
+            address: newVoter.address,
+            district: newVoter.district,
+            householdNo: newVoter.householdNo,
+            gramaNiladhari: newVoter.gramaNiladhari,
+            password: hashedPassword
+        };
+
+        data:VoterInsert[] voterInsertArr = [voterInsert];
+        string[] | error response = dbClient->/voters.post(voterInsertArr);
+
+        if response is error {
+            return response;
+        }
+        return http:CREATED;
+    }
+
+    // Voter Login
+    @http:ResourceConfig {}
+    resource function post voters/login(@http:Payload VoterLogin loginDetails)
+    returns http:Response|http:Unauthorized|error {
+    
+        
+        sql:ParameterizedQuery query = `SELECT * FROM "Voter" WHERE "nationalId"::text = ${loginDetails.nationalId}`;
+        
+        stream<data:Voter, persist:Error?>|error queryResult = dbClient->queryNativeSQL(query);
+        
+        if queryResult is error {
+            return error("Internal Server Error");
+        }
+        
+        stream<data:Voter, persist:Error?> voterStream = queryResult;
```

Comment:
```suggestion
```

Reply:
Once you check whether `voterStream` is an error and it returns in a negation, Ballerina treats `voterStream` as of type `stream<data:Voter, persist:Error?>` afterwards. So you don't have to explicitly assign to another variable